setup.py

```diff
@@ -50,5 +50,6 @@
             'Topic :: Software Development :: Libraries :: Python Modules',
             ],
     zip_safe=False,
-    include_package_data=True
+    include_package_data=True,
+    python_requires=">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*",
```

Comment:
This is a tough one: are we sure xlrd doesn't work with 3.0-3.3?

Reply:
Per 02b74857b4056c62014290cc7dd7a50f81ccf51a xlrd no longer supports Python 3.3. Using `python_requires` allows pip to install the next compatible release on a Python 3.3 virtualenv, pip will use the next highest compatible version instead. If it is unknown if xlrd is compatible with Python 3.3 by chance, I think it is better to steer users to a version that we know is compatible and tested. As xlrd upgrades and removes old support, users won't mistakenly upgrade to incompatible versions.

Reply:
"supports" and "works with" are subtly different though; per that commit we stopped testing on that version, but I'm not aware of any problems with the current codebase on 3.3. That said, I'm not sure how many people would even be using 3.3 ;-)

What I don't want it to prevent pip installing a version which *may* work even though it's not tested. 

Thoughts?

Reply:
Here are the pip stats for the last 30 days:

```
$ pypinfo --percent --pip xlrd pyversion
| python_version | percent | download_count |
| -------------- | ------- | -------------- |
| 2.7            |   62.1% |        252,870 |
| 3.6            |   16.0% |         65,039 |
| 3.4            |   10.9% |         44,252 |
| 3.5            |   10.7% |         43,570 |
| 2.6            |    0.2% |            918 |
| 3.7            |    0.1% |            231 |
| 3.3            |    0.1% |            221 |
| None           |    0.0% |              5 |
| 3.2            |    0.0% |              3 |
```

So I would say very few as a percentage of total users.

> What I don't want it to prevent pip installing a version which may work even though it's not tested.

I see what you're saying.

I suppose what I want is to prevent pip from installing a version that _may not_ work. It may not work because it is untested and unverified. For those users, I'd prefer to have pip install a version that we know works. I think it would ultimately lead to fewer surprises to users trying to install a working version of xlrd.

Reply:
With those stats, I think we're good to go.